msal/src/main/java/com/microsoft/identity/client/ISingleAccountPublicClientApplication.java

```diff
@@ -72,6 +75,18 @@
          * @param currentAccount the current signed-in account. This could be nil.
          * */
         void onAccountChanged(final IAccount priorAccount, final IAccount currentAccount);
+    }
+
+    /**
+     * Callback for asynchronous loading of broker AccountRecord account (in Single account mode).
+     */
+    interface SingleAccountLoadedCallback {
 
+        /**
+         * Called once the signed-in account (if there is any), has been loaded from the broker.
+         * @param accountRecord The accountRecord in broker. This could be null.
+         */
+        void onAccountLoaded(@Nullable final AccountRecord accountRecord);
```

Comment:
Just verifying that this is not intended for public consumption, right?

Reply:
(as in, `AccountRecord` is an internal class and shouldn't be exposed to users of the library directly) - just clarifying intent here.... can't recall if inner-interfaces of `public interfaces` are implicitly `public` or not

Reply:
Yeah, it looks like this would be implicitly `public` from this post: https://beginnersbook.com/2016/03/nested-or-inner-interfaces-in-java/

>Nested interfaces declared inside class can take any access modifier, however nested interface declared inside interface is public implicitly.

Reply:
This is a callback function that is used by BrokerMsalController. User shouldn't need this.

As the concept of package-private interface doesn't exist in java, here are the things I'd propose.

1. Rename the function and add the description that this is for internal use only.
2. Move this out of the ISingleAccount interface, and put it in the actual object itself. (So that the interface would only contains dev-exposing interfaces.

What do you think?

Reply:
#2 sounds good to me